### PR TITLE
deps!: Bumps go-header

### PR DESCRIPTION
**Pull request by renaynay**
Bumps go-header - includes breaks to `Getter` interface + metrics instantiation for header pkg

### CODE REVIEW

**Review by vgonkivs — APPROVED** [review]

File: nodebuilder/header/config.go [hunk 1]
```diff
@@ -31,6 +31,9 @@ type Config struct {
 
 	Server p2p_exchange.ServerParameters
 	Client p2p_exchange.ClientParameters `toml:",omitempty"`
+
+	// enables metrics for the entire header module
+	MetricsEnabled bool
```
Comment (Wondertan) [inline nodebuilder/header/config.go:36]: I would approve instantly if we do global func

Comment (renaynay) [reply nodebuilder/header/config.go:36]: global func or var?

Comment (Wondertan) [reply nodebuilder/header/config.go:36]: changed to var

Comment (renaynay) [reply nodebuilder/header/config.go]: https://github.com/celestiaorg/celestia-node/pull/2844/commits/1bda27f371774ef1da20b79f25deeaa1e2ab3d5c

image

**Review by Wondertan — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by liamsi — APPROVED** [review]

**Pull request merged**